testnet/bns.js

```diff
@@ -0,0 +1,354 @@
+      console.log('claimSatoshis', claimSatoshis);
+      console.log('changeAddress', changeAddress);
+      console.log('changeSatoshis', changeSatoshis);
+      console.log('isTransform', new Bool(false));
+      console.log('issuerSig', issuerSig);
+      console.log('issuerPubKey', issuerPubKey);
+      return tree.extend(
+        preimage,
+        dividedSatoshisBytesWithSize,
+        claimSatoshis,
+        changeAddress,
+        new Bytes(changeSatoshis),
+        new Bool(false),
+        issuerSig,
+        issuerPubKey).toScript()
+    })
+
+   /* extendRootTx.addInput(new bsv.Transaction.Input({
+      prevTxId: utxo.txId,
+      outputIndex: utxo.outputIndex,
+      script: ''
+    }), utxo.script, utxo.satoshis); */
+
+    // NFT claim output
+    extendRootTx.setOutput(0, (tx) => {
+      const deployData = buildNFTPublicKeyHashOut(num2bin(0, 36), privateKey.toAddress().toHex().substring(2))
+      return new bsv.Transaction.Output({
+        script: deployData,
+        satoshis: claimSatoshisInt,
+      })
+    })
+    currentDimension++;
+    for (let i = 0; i < letters.length; i++) {
+      let letter = letters[i];
+      const combinedDupHash = prevDupHash + 'ff'; // The parent root node is '00' 
+      const dupHash = bsv.crypto.Hash.ripemd160(Buffer.from(combinedDupHash, 'hex')).toString('hex');
+      const newLockingScript = getLockingScriptForCharacter(letter, currentDimension, dupHash);
+      const lockingScriptSizeNew = newLockingScript.toHex().length / 2;
+
+      if (i  == 0) {
+        //console.log('combinedDupHash', prevDupHash, combinedDupHash, dupHash);
+        console.log('tree before', tree.lockingScript.toASM());
+        console.log('lockingScriptSizeNew size', lockingScriptSizeNew, newLockingScript.toASM(), newLockingScript.toHex(), num2bin(lockingScriptSizeNew, 2));
+      }
+     
+      step2ExtendLockingScripts.push({
+        newLockingScript,
+        dupHash
+      });
+
+      extendRootTx.setOutput(i + 1, (tx) => {
+        return new bsv.Transaction.Output({
+          script: newLockingScript,
+          satoshis: dividedSatsResult,
+        })
+      })
+    }
+
+    // Add Change
+   /* changeSatoshis = (utxo.satoshis - FEE - (dividedSats));
+    console.log('changeSatoshis', changeSatoshis);
+    const changeOutputScript = bsv.Script.buildPublicKeyHashOut(privateKey.toAddress())
+    // const changea4187bd7b8a126716eeb9586eeb1261d5861d24c = bsv.Script.buildPublicKeyHashOut('a4187bd7b8a126716eeb9586eeb1261d5861d24c');
+    extendRootTx.addOutput(new bsv.Transaction.Output({
+      script: changeOutputScript,
+      satoshis: changeSatoshis
+    })); */
+    
+    //unlockP2PKHInput(privateKey, extendRootTx, 1, Signature.SIGHASH_ALL | Signature.SIGHASH_ANYONECANPAY | Signature.SIGHASH_FORKID);
+
+
+  //  extendRootTx.from(await fetchUtxos(privateKey.toAddress()))
+   extendRootTx.change(privateKey.toAddress())
+   .sign(privateKey)
```

Comment:
Some reason the tx fails about 50% of the time

Reply:
try to bump scryptlib to 0.10.4, which fix this issue: https://github.com/sCrypt-Inc/scryptlib/issues/156